actors/runtime/src/main/java/cloud/orbit/actors/runtime/Hosting.java

```diff
@@ -433,13 +452,16 @@ private RemoteKey createRemoteKey(final RemoteReference actorReference)
                 return Task.fromException(new UncheckedException(timeoutMessage));
             }
 
-            // read volatile field into local field
+            // read volatile fields into local fields
             final List<NodeInfo> currentServerNodes = serverNodes;
+            final Set<String> currentTargetPlacementGroups = targetPlacementGroups;
 
             // filter out nodes which cannot host actors (nodes which are not running and nodes which cannot host this actor)
+            // and nodes that are not in any target placement group
             final List<NodeInfo> potentialNodes = currentServerNodes.stream()
                     .filter(n -> (!n.cannotHostActors && n.state == NodeState.RUNNING)
-                            && actorSupported_no != n.canActivate.getOrDefault(interfaceClassName, actorSupported_yes))
+                            && actorSupported_no != n.canActivate.getOrDefault(interfaceClassName, actorSupported_yes)
+                            && (n.placementGroupPending.get() || currentTargetPlacementGroups.contains(n.placementGroup)))
```

Comment:
This should be `(!n.placementGroupPending.get() && currentTargetPlacementGroups.contains(n.placementGroup))`

Reply:
Oops, yep.

Reply:
Actually no, not really (at first glance I though this was the filter down below. What I'm doing here is getting nodes that we either haven't read their placement group yet (`placementGroupPending == true`), or we know they're in a target placement group. The ones whose placement group is still unknown will be updated in the loop below.

Reply:
Ok, that makes sense. Then I guess this is all good.